src/init/variables/VarResolver.ts

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable tsdoc/syntax */
+import yargs from 'yargs';
+import { createErrorMessage } from '../..';
+import { AsyncHandler } from '../../util/handlers/AsyncHandler';
+import { modulePathPlaceholder } from '../../util/PathUtil';
+import type { VarComputer } from './VarComputer';
+
+const defaultConfig = `${modulePathPlaceholder}config/default.json`;
+const defaultVarConfig = `${modulePathPlaceholder}config/app-setup/vars.json`;
+
+export type YargsArgOptions = Record<string, yargs.Options>;
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export function setupYargvWithBaseArgs(yargv: yargs.Argv) {
+  // Values from componentsjs LogLevel type
+  const logLevels = [ 'error', 'warn', 'info', 'verbose', 'debug', 'silly' ];
+  return yargv.options({
+    /**
+   * CLI options needed for performing meta-process of app initialization.
+   * These options doesn't contribute to components-js vars normally.
+   */
+    config: { type: 'string', alias: 'c', default: defaultConfig, requiresArg: true },
+    loggingLevel: { type: 'string', alias: 'l', default: 'info', requiresArg: true, choices: logLevels },
```

Comment:
```suggestion
   // CLI options needed for performing app initialization
  return yargv.options({
    config: { type: 'string', alias: 'c', default: defaultConfig, requiresArg: true },
    loggingLevel: { type: 'string', alias: 'l', default: 'info', requiresArg: true, choices: logLevels },
```

Reply:
But does `loggingLevel` really need to be initialized here? I thought that all went through Components.js?

Reply:
This loggingLevel is for `componentsjs` itself too. As mentioned in doc-strings, these definitions are for the `meta-process`.

Reply:
Okay, that makes sense.